### PR TITLE
WIP: Add the right hierarchy of Breadcrumbs to pages 

### PR DESCRIPTION
**Pull request by Agaba-derrick**
This pull request addresses issue #652 and proposes enhancements to the Breadcrumb component to support a more generic usage. The changes enable the component to accept an array of objects and generate a full hierarchy of breadcrumbs as needed, while retaining the default "Home" breadcrumb.
Screenshot from 2024-02-04 23-43-25
Screenshot from 2024-02-05 00-55-49
Screenshot from 2024-02-05 00-56-15

### CODE REVIEW

**Comment by Agaba-derrick** [comment]
@mozzy11  requesting for a review

**Review by tendomart — COMMENTED** [review]
Thanks @Agaba-derrick , where is the re-usable component though ?

I can't seemt to locate it here ?

**Comment by Agaba-derrick** [comment]
> Thanks @Agaba-derrick , where is the re-usable component though ?
> 
> I can't seem to locate it here ?

I'm on it in a few thank you !!!!

**Comment by mozzy11** [comment]
@Agaba-derrick , The breadcrumb should be the first item on the page ,since its a navigational element. 
And als the indention of the breadcrupms  should be consistent with other bread crumps. 
Its usually wrapped with in a Grid element to give it an appropriate indention  ie

 ```
<Grid fullWidth={true}>
        <Column lg={16}>
          <Breadcrumb>
            <BreadcrumbItem href="/">
              {intl.formatMessage({ id: "home.label" })}
            </BreadcrumbItem>
            <BreadcrumbItem href="/ImmunohistochemistryDashboard">
              {intl.formatMessage({
                id: "immunohistochemistry.label.dashboard",
              })}
            </BreadcrumbItem>
          </Breadcrumb>
        </Column>
      </Grid>

```

Screenshot 2024-02-05 at 14 09 41

**Comment by mozzy11** [comment]
@Agaba-derrick , Bread Crumps basically help link back to the parent pages with in the Navigation Hierachy of the Application  . From the screen shots you've attached , ie the Pathogoy dashboard page has `Home/Pathology`. yet it only has one parent Page ie` Home`
 in its Navigation Hierachy

**Comment by Agaba-derrick** [comment]
> @Agaba-derrick , Bread Crumps basically help link back to the parent pages with in the Navigation Hierachy of the Application . From the screen shots you've attached , ie the Pathogoy dashboard page has `Home/Pathology`. yet it only has one parent Page ie` Home` in its Navigation Hierachy

@mozzy11 Thanks, let me work on it

**Comment by Agaba-derrick** [comment]
> > @Agaba-derrick , Bread Crumps basically help link back to the parent pages with in the Navigation Hierachy of the Application . From the screen shots you've attached , ie the Pathogoy dashboard page has `Home/Pathology`. yet it only has one parent Page ie` Home` in its Navigation Hierachy
> 
> @mozzy11 Thanks, let me work on it

noted

File: frontend/src/components/utils/store/index.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { create } from 'zustand'
+
+const availableStores = {}
+
+export function createGlobalStore(name, initialState) {
+    const existingStore = availableStores[name];
```
Comment (mozzy11) [inline frontend/src/components/utils/store/index.js:6]: @Agaba-derrick am not sure what you are trying to do here but looks like you're making this unnecessarily complicated and I think it should be simple

**Pull request closed without merging**